### PR TITLE
wgrad-accumulation patch for legacy megatron-LM

### PR DESCRIPTION
**Pull request by ksivaman**

### CODE REVIEW

File: transformer_engine/pytorch/module/layernorm_linear.py [hunk 1]
```diff
@@ -473,14 +473,20 @@ def backward(
             if not ctx.use_bias:
                 grad_bias = None
 
-        # Handle custom DDP from mcore.
-        weight.grad_added_to_main_grad = ctx.fuse_wgrad_accumulation
+        if weight.requires_grad:
+            # Handle custom DDP from mcore.
```
Comment (deepakn94) [inline transformer_engine/pytorch/module/layernorm_linear.py:477]: Maybe create a smaller helper method that takes in `weight` and `fuse_wgrad_accumulation` arguments? Then you don't need these nested if conditions everywhere.

**Review by deepakn94 — APPROVED** [review]
Looks good, small comment.

**Review by timmoon10 — APPROVED** [review]

**Pull request merged**